### PR TITLE
allow switching to a different video while processing is in progress

### PR DESCRIPTION
**Pull request by adennie**
note: this branch is derived from the remove-licensing branch, so it contains that branch's commits

### CODE REVIEW

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/MainActivity.java [hunk 1]
```diff
@@ -16,19 +16,11 @@
...
-
-import java.util.List;
-
 /*
 A sample app showing how to use VideoFileDetector. This detector must be run on a non-Looper thread.
+*/
 
-This app is not a release version and is known to have bugs. Specifically, if the user tries to load a video while the Detector is processing, the app will crash.
- */
-
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/MainActivity.java]: remove unused implementation of ImageListener

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/MainActivity.java [hunk 2]
```diff
@@ -50,17 +42,20 @@ void processVideo(String filename) {
         videoThread.start();
     }
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/MainActivity.java]: change XML-bound callback method name to camelCase for consistency

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java [hunk 3]
```diff
@@ -13,18 +13,19 @@
 
 /**
  * A thread to manage the VideoDetector.
- * <p/>
+ * <p>
  * Note: This is required since running the VideoDetector in the main thread will crash the application.
  */
 public class VideoDetectorThread extends Thread implements Detector.ImageListener {
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java]: privatize

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java [hunk 4]
```diff
@@ -47,17 +46,28 @@ public void run() {
         } catch (Exception e) {
             Log.e(LOG_TAG, e.getMessage());
         } finally {
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java]: note: in abort scenario, detector will already be stopped at this point

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java [hunk 5]
```diff
@@ -47,17 +46,28 @@ public void run() {
...
-            detector.stop();
+            if (detector.isRunning()) {
+                detector.stop();
+            }
+            // notify waiting threads that we're done
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java]: signal to waiting abort() method

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java [hunk 6]
```diff
@@ -47,17 +46,28 @@ public void run() {
...
+            }
         }
     }
 
-
     @Override
     public void onImageResults(List<Face> list, Frame image, final float timestamp) {
 
         final Frame frame = image;
         final List<Face> faces = list;
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java]: each time image results come back, check for outstanding abort request.  If requested, stop processing.

File: VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java [hunk 7]
```diff
@@ -161,8 +171,26 @@ public void run() {
         });
     }
 
+    /**
```
Comment (adennie) [inline VideoDetectorDemo/src/main/java/com/affectiva/videodetectordemo/VideoDetectorThread.java]: let the main thread request an abort

**Comment by mostafazh** [comment]
Looks good, let's merge this after we merge #11 and #12

**Pull request merged**